packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -52,6 +52,13 @@ typedef ScrollableWidgetBuilder = Widget Function(
 /// fire when [pixels] changes without [size] changing. For example, if the
 /// constraints provided to an attached sheet change.
 class DraggableScrollableController extends ChangeNotifier {
+  /// Creates a controller for [DraggableScrollableSheet].
+  DraggableScrollableController() {
+    if (kFlutterMemoryAllocationsEnabled) {
```

Comment:
I am less familiar here, what is `kFlutterMemoryAllocationsEnabled`? Will it be on by default in the future? If so, are there plans to come back and remove this?

Reply:
The `kFlutterMemoryAllocationsEnabled` is true only in the debug mode. I think setting it to true by default is not needed in production and won't be in the future. In the future, creation/disposal dispatching will be separated into `Disposable` mixin. You can read about future plans here: https://docs.google.com/document/d/1cjylBOczGk70x-sn1iZYVNAfpPZdtGWsm_KzoNQX0jc/edit#heading=h.b5wf7efqvos

@polina-c, please correct me if I'm wrong.

Reply:
True, this flag is only on, when leak tracking is needed. 
By default it is on only in debug mode, to tree shake leak tracking code and to have better performance. When needed the flag can be enabled for release and profile mode by compiling with this flag. 

ChangeNotifier is special disposable, because it is mixin and thus does not have constructor.
See more details in documentation for maybeDispatchObjectCreation: https://github.com/flutter/flutter/blob/a11b21dba87c987fc004d2aaace604b82ec6c831/packages/flutter/lib/src/foundation/change_notifier.dart#L230

Does it help?

Reply:
This will go away if we can enable object tacing without taking stack trace: https://docs.google.com/document/d/1cjylBOczGk70x-sn1iZYVNAfpPZdtGWsm_KzoNQX0jc/edit#bookmark=id.vp85epkbqmp9

Reply:
We need to add maybeDispatchObjectCreation, because DraggableScrollableController leaks in some tests and it is not possible to detect who is owner, that created, but did not dispose the object.

Reply:
Ah ok! So eventually in the future we won't need to include kFlutterMemoryAllocationsEnabled in every ChangeNotifier like this? It made me think, hm this should probably be in a lower level of the framework by default so we don't have to remember to include it every time we create a ChangeNotifier.

Reply:
Yes, this suggestion is exactly to address this at lower level: https://docs.google.com/document/d/1cjylBOczGk70x-sn1iZYVNAfpPZdtGWsm_KzoNQX0jc/edit#bookmark=id.vp85epkbqmp9

If we forget to include the invocation to the child of ChangeNotifier, it is not big deal, because it can be added after, per instruction for leak troubleshooting, when it is hard to detect who is owner: https://github.com/dart-lang/leak_tracker/blob/main/doc/TROUBLESHOOT.md#1-it-is-not-clear-who-owns-changenotifier